### PR TITLE
refactor(efb/mcdu): move simbrief username input out of mcdu

### LINKED ISSUE #5725: efb: Fix SimBrief Setting
### Aircraft Version

Development

### Description

The SimBrief setting should work the same as the MCDU (accept username or pilot ID, store pilot ID), and use the same key in NXDatastore (`CONFIG_SIMBRIEF_USERID`). Then the option can be removed from the MCDU as it is one of the last 3 remaining after #5344.

### References (optional)

_No response_

### Additional info (optional)

_No response_

### Discord Username (optional)

_No response_

### PR DESCRIPTION
**Pull request by ErickSharp**
Fixes #5725, #5761

## Summary of Changes

Moves the simbrief username input out of the MCDU and into the EFB.

## Screenshots (if necessary)

image
image
image
image

## Additional context

Discord username (if different from GitHub): Erick [Z-6]#6484

## Testing instructions

Go into the EFB settings then ATSU/AOC then make a change to the simbrief username input then unfocus it and it will try to set your username to the new value, if it is unable to, you should see a 'Simbrief Error' message on the left for about 4 seconds.

Ensure that data loading from simbrief works as it did before.

Ensure that the text input fields in the EFB work as they did before.

## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, click on the bottom **PR** tab
1. Click on the **A32NX** download link at the bottom of the page

### CODE REVIEW

**Comment by jaoord** [comment]
Does this also fix #5761 ?

**Comment by ErickSharp** [comment]
> Does this also fix #5761 ?

Yes, I think it should. Thank you for mentioning the issue ;)

File: src/instruments/src/EFB/Components/Form/SimpleInput/SimpleInput.tsx [hunk 1]
```diff
@@ -105,7 +114,7 @@ const SimpleInput: FC<SimpleInputProps> = (props) => {
                         <input
                             className={`px-5 py-1.5 text-lg text-gray-300 rounded-lg bg-navy-light border-2 border-navy-light focus-within:outline-none
                             focus-within:border-teal-light-contrast ${props.className}`}
-                            value={displayValue}
+                            value={props.value ?? displayValue}
```
Comment (tracernz) [inline src/instruments/src/EFB/Components/Form/SimpleInput/SimpleInput.tsx:117]: Can you explain the reason for this one? If `props.value` changes the useEffect above would update `displayValue`, right? I'm probably missing something.

File: src/instruments/src/EFB/Settings/Settings.tsx [hunk 2]
```diff
@@ -358,13 +417,20 @@ const ATSUAOCPage = (props: {simbriefUsername, setSimbriefUsername}) => {
                 </SelectGroup>
             </div>
             <div className="py-4 flex flex-row justify-between items-center">
-                <span className="text-lg text-gray-300">Simbrief Username</span>
+                <span className="text-lg text-gray-300">
+                    Simbrief Username
```
Comment (tracernz) [inline src/instruments/src/EFB/Settings/Settings.tsx:421]: Sim**B**rief
Should this be `SimBrief Username/Pilot ID`?

File: src/instruments/src/EFB/Settings/Settings.tsx [hunk 3]
```diff
@@ -358,13 +417,20 @@ const ATSUAOCPage = (props: {simbriefUsername, setSimbriefUsername}) => {
...
+                    <span className={`${!simbriefError && 'hidden'} text-red-600`}>
+                        <span className="text-white"> | </span>
+                        Simbrief Error
```
Comment (tracernz) [inline src/instruments/src/EFB/Settings/Settings.tsx:424]: Sim**B**rief

**Review by tracernz — COMMENTED** [review]
Thank you for picking this up! 😁

File: src/instruments/src/EFB/Components/Form/SimpleInput/SimpleInput.tsx (hunk 1 above)
Comment (ErickSharp) [reply src/instruments/src/EFB/Components/Form/SimpleInput/SimpleInput.tsx]: Sorry about this one, I think I was testing something and accidentally forgot to remove it before committing.

Comment (tracernz) [reply src/instruments/src/EFB/Components/Form/SimpleInput/SimpleInput.tsx]: I do that all the time. 😁

**Review by tracernz — APPROVED** [review]

**Comment by Benjozork** [comment]
Should probably test the OFP loading in the EFB as well as part of the reports.

**Comment by ewold** [comment]
> Should probably test the OFP loading in the EFB as well as part of the reports.

Good point. retested and added to the report.

**Comment by TheSquirtingPantin** [comment]
Quality Assurance Tester

Discord: Remi#8871
Object of testing: #5813
Tier of Testing: 1
Date: 17/09/2021

Testing Process:
1) Started on RWY, with airplane set for takeoff. 
2) Checked logic/entries in the AOC/ATSU menu
3) Loaded a generated simbrief OFP.
4) Verified entries are correct in MCDU and EFB OFP. 
5) Loaded another simbrief generated OFP using different user ID (both with active OFP and inactive OFP)
6) Verified correct loading in both EFB and MCDU


Negatives: None

Conclusions: PASS. PR is working as expected. Good logic for ID box in the EFB ATSU/AOC menu. OFP is loaded and displayed correctly in both MCDU and EFB. Good job !

**Comment by frankkopp** [comment]
Quality Assurance Tester Report

Discord: Cdr_Maverick#6475
Object of testing: #5813
Tier of Testing: 1
Date: 18.9.21

Testing Process: 
C&D, checking MCDU settings, checking EFB settings, testing invalid simBrief account, testing ID
Testing import on MCDU and verify OFP data in MCDU
Testing simBrief EFB import
Restarting sim to check if persisted

Negatives: None 

Testing Results: Passed

Conclusions: LGTM - will update EFB and Beginner Guide docs and prepare a PR for it

Media:

**Comment by jcalor** [comment]
Quality Assurance Trainee Report

Discord          : Jorge#6816
Object of testing: #5813 
Tier of Testing  : 1
Date             : 18/09/2021

Testing Process: 
- Spawned C&D at LFPG
- Check my Simbrief ID on flyPad - is correct (Screenshot 1)
- Did a test OFP on Simbrief webpage (LFPG-EDDM). Checked on MCDU if it imports Simbrief data - it does. (Screenshot 2)
- Set a random Simbrief ID to see if Simbrief Error appears - it does and it set my correct ID (see GIF)
- Restarted sim to see if it still works - it does
Negatives:       
N/A

Testing Results: Passed

Conclusions: LGTM - No problems at my own, good job

Media:       
Screenshot 1
https://prnt.sc/1sot4n9

Screenshot 2
https://prnt.sc/1sot884

GIF
https://i.gyazo.com/10458154e0364d54b348d238476286da.mp4

**Comment by ShadowFrost3** [comment]
Quality Assurance Trainee Report


Object of testing: #5813
Tier of Testing:1
Date: 18/09/2021

Testing Process: 
C&D load check EFB, check EFB simbrief error, simbrief number, simbrief username
Verified both EFB and MCDU could low simbrief flight plan
Negatives: 
Testing Results: Passed

Conclusions: Works as expected, provides errors when username is incorrect and correctly inputs the number form of a username. 
Media: 
image

**Pull request merged**